### PR TITLE
feature(breeding): allow sorting by and displaying total hatches

### PR DESCRIPTION
**Pull request by SpenserJ**
This was requested by a few people on Discord so that they can breed their pokemon an equal number of times.
image

### CODE REVIEW

**Review by RedSparr0w — APPROVED** [review]
Looks good to me 👍 

Do you think `Times Hatched` would make more sense than `Total Hatches`?
I'll leave it up to you to decide and merge though.

**Comment by KrushGames** [comment]
maybe you guys could do a poll in the voting channel on discord? that way more people can get an opinion on it?

File: src/components/breeding.html [hunk 1]
```diff
@@ -45,6 +45,7 @@ <h5 class="modal-title" id="breedingModalLabel" data-bind="tooltip: { title: `Yo
                                     <option value="attackBonus">Attack Bonus</option>
                                     <option value="baseAttack">Base Attack</option>
                                     <option value="eggSteps">Egg Steps</option>
+                                    <option value="totalHatches">Total Hatches</option>
```
Comment (RedSparr0w) [inline src/components/breeding.html:48]:
```suggestion
                                    <option value="timesHatched">Times Hatched</option>
```

File: src/scripts/breeding/BreedingController.ts [hunk 2]
```diff
@@ -205,6 +205,7 @@ class BreedingController {
             case 'attackBonus': return `Attack Bonus: ${Math.floor(pokemon.baseAttack * (GameConstants.BREEDING_ATTACK_BONUS / 100)).toLocaleString('en-US')}`;
             case 'baseAttack': return `Base Attack: ${pokemon.baseAttack.toLocaleString('en-US')}`;
             case 'eggSteps': return `Egg Steps: ${App.game.breeding.getSteps(pokemonData.eggCycles).toLocaleString('en-US')}`;
+            case 'totalHatches': return `Hatches: ${App.game.statistics.pokemonHatched[pokemonData.id]() || 0}`;
```
Comment (RedSparr0w) [inline src/scripts/breeding/BreedingController.ts:208]:
```suggestion
            case 'timesHatched': return `Hatches: ${App.game.statistics.pokemonHatched[pokemonData.id]() || 0}`;
```

File: src/scripts/settings/SortOptions.ts [hunk 3]
```diff
@@ -7,6 +7,7 @@ enum SortOptions {
     'baseAttack' = 5,
     'breedingEfficiency' = 6,
     'eggCycles' = 7,
+    'totalHatches' = 8,
```
Comment (RedSparr0w) [inline src/scripts/settings/SortOptions.ts:10]:
```suggestion
    'timesHatched' = 8,
```

File: src/scripts/settings/SortOptions.ts [hunk 4]
```diff
@@ -65,4 +66,9 @@ const SortOptionConfigs: Record<SortOptions, SortOptionConfig> = {
         'text': 'Egg Steps',
         'getValue': p => pokemonMap[p.name].eggCycles,
     },
+
+    [SortOptions.totalHatches]: {
+        'text': 'Total Hatches',
```
Comment (RedSparr0w) [inline src/scripts/settings/SortOptions.ts:71]:
```suggestion
    [SortOptions.timesHatched]: {
        'text': 'Times Hatched',
```

**Review by RedSparr0w — APPROVED** [review]
Updating from `Total Hatches` → `Times Hatched` based on Discord poll:
image

**Pull request merged**